From 91608e885e7809909c25dac2b07e2405bacbf0a5 Mon Sep 17 00:00:00 2001
From: Mattias Busck <mattias.busck@hostmobility.com>
Date: Wed, 12 Oct 2022 09:48:56 +0200
Subject: [PATCH] Set Linux DT file to hmx1 if u-boot detects legacy

Update u-boot environment settings to set device tree to
imx8mp-var-dart-hmx1.dtb if the Variscite carrier board eeprom cannot be
read and fdt_file has not been set.
---
 include/configs/imx8mp_var_dart.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/imx8mp_var_dart.h b/include/configs/imx8mp_var_dart.h
index d2a3e43880..93b76196c9 100644
--- a/include/configs/imx8mp_var_dart.h
+++ b/include/configs/imx8mp_var_dart.h
@@ -122,7 +122,7 @@
 				"setenv fdt_file imx8mp-var-som-symphony.dtb; " \
 			"else " \
 				"if test $carrier_rev = legacy; then " \
-					"setenv fdt_file imx8mp-var-dart-dt8mcustomboard-legacy.dtb;" \
+					"setenv fdt_file imx8mp-var-dart-hmx1.dtb;" \
 				"else " \
 					"setenv fdt_file imx8mp-var-dart-dt8mcustomboard.dtb;" \
 				"fi; " \
-- 
2.30.2

